scripts/prepare_sector_network.py

```
@@ -1306,6 +1324,104 @@ def add_storage(n, costs):
         y=n.buses.loc[list(spatial.nodes)].y.values,
     )
 
+    # Existing battery capacity per AC node (MW) passed in
+    if existing_battery_capacity is None:
+        existing_battery_capacity = {}
+
+    total = float(sum(existing_battery_capacity.values()))
+    nodes_with_cap = len([p for p in existing_battery_capacity.values() if p > 0])
+    logger.info(
+        f"Battery carry-over from elec.nc: {total/1e3:.2f} GW across {nodes_with_cap} nodes."
+    )
+
+    try:

```

What are particular reasons of having `try-catch` block here?

Generally, using of try-catch blocks is discouraged unless there is a strong need to address some special cases which can happen as a part of a normal workflow [e.g. some url source won't be available, or data field doesn't present in the dataset]. In those cases a type of the exception must be specified as otherwise the errors can be masked

---

This should mask issues related to malformed or missing powerplant metadata, but it's maybe too careful to have it here. I will remove it